### DISCUSSION TITLE
How to create a post template which fetches posts by tag

### DISCUSSION THREAD

**Discussion by 3muchfun** [GetPublii/Publii#1465] [Help] [answered]
Hi, 
in the old forum @bobmitro wrote how to change a post to a page:
https://forum.getpublii.com/topic/post-vs-page/:
_Mark a post as Hidden, and then it will work as a “static” page._

My question is: how I make the 'page' to be shown on the front page within the list of posts ?

Secondly: My goal is to create a 'page' that holds links to other posts, e.g.:

- front page
    |_  page
.            |_ post 1
.            |_ post 2
.            |_ post 3
.            |_ ...
    
and this 'page' should behave like the front page ... I use the Mercury free theme
Thanks in advance, Michael

**Comment by candidexmedia** [comment]
Hi @3muchfun!

I admit, I'm a little confused by this post, but hopefully this answer helps...

> My question is: how I make the 'page' to be shown on the front page within the list of posts ?

Don't mark the post as hidden.<sup>1</sup>

> Secondly: My goal is to create a 'page' that holds links to other posts, e.g.:

It's possible to set an existing post/page<sup>1</sup> as your site's home page.

![image](https://github.com/GetPublii/Publii/assets/55474996/f54dd18f-f240-4fe5-83dc-cf3b9e0bff34)

Instead of a listing of posts, it'll display your selected post.

That said, if you want to keep Publii's automated "Latest posts" listings _and_ have some kind of opening text above it that displays other links, use the WYSIWYG text editor in the "Hero section" theme setting:

![image](https://i.imgur.com/RLbd8i1.png)

You can then link to other pages from there using the link button.

![image](https://i.imgur.com/mV2aAHT.png)

Let me know if that clarifies things. If not, could you sketch out what you're hoping to achieve?

-----
<sup>1</sup> For the time being, "posts" and "pages" (in this context) are more or less the same. Hiding a post doesn't change/convert it into a page -- it just hides the posts from listings.

**Reply by 3muchfun** [reply]
Ok, I try to explain it another way: 
Assuming, I have a site fluegge.net/unterwegs, which holds several posts (to galleries). I now want to create a post '_Mallorca_', which have links to 4 other posts (each with galleries). And I want this post '_Mallorca_' behave like and have the styles like the front page.
In principle, a tag page would do it, but I have no idea, how to get a tag page listed like a post in the list of posts on the front page...

**Reply by candidexmedia** [reply]
> And I want this post 'Mallorca' behave like and have the styles like the front page.

Do you have a visual of the style you want? I'm assuming it's the filter and post cards like the front page of Mercury, but I want to make sure we're on the same page.

Do you want to fetch all posts, all posts under a certain tag, or just selected posts?

I suspect that you'll need to create a new post template in your site folder using a text editor, copy some index.hbs content over to the new template, and adjust some of the handlebars ([essentially what's described here](https://cv.domagic.site/posts/alternative-post-template-with-filter-for-publii-cms)). That said, I want to make sure I understand your request correctly.

**Reply by 3muchfun** [reply]
_Do you want to fetch all posts, all posts under a certain tag, or just selected posts?_
-> I want to show all posts related to one tag

I made copy of the index.hbs (of the Mercury theme) as _post-mallorca.hbs_ and followed Yaroslav's instructions. It worked pretty well except one thing: I found no way to set the filter to a certain tag, #11 in my case, or as slug 'mallorca'. I think it should be GetPostsByTags, but ... I want this filter hard coded, not dynamically.
Any idea?
Thanks in advance

**Reply by candidexmedia** [reply]
Can you share your code for `post-mallorca.hbs`?

**Reply by 3muchfun** [reply]
here you are: 
https://fluegge.net/ohnein/post-mallorcahbs/

**Reply by 3muchfun** [reply]
All I changed from the index.hbs is:
change
`{{#each posts}}`
to
`{{#each @website.contentStructure.posts}}`

**Reply by 3muchfun** [reply]
And I created a Tag Mallorca with Id 11, so I thought '{{#getPostsByTags 8 11}}' would do the trick, but I don't know, where to put it ...

**Reply by candidexmedia** [reply, marked as answer]
This might be the line to modify:
 ```handlebars
{{#each @website.contentStructure.posts}}
 ```
I did some digging, and you might have to use a different helper since you're no longer on the home page index. Give this article a look: https://getpublii.com/dev/how-to-display-posts-connected-with-a-specific-tag-name/

------
Sidenote: to properly format codeblocks in GitHub, use ` ``` ` at the top and bottom of the code. You can even specify the code language for highlighting in the top ` ``` `

For example:

 ```
| ```html
| paste code here
| ```
 ```

**Reply by 3muchfun** [reply]
That does the trick, thank you :-)
I changed `{{#each @website.contentStructure.posts}}` to `{{#each @website.contentStructure.tags}}` and and added 
```
{{#checkIf this.name '==' 'TAGNAME'}}
    {{#if this.postsNumber}}
    {{#each this.posts}}
```
as described in the link you provided.

You can see the result here, if you like: https://fluegge.net/unterwegs/  and choose the card for _Mallorca_

**Reply by candidexmedia** [reply]
That is wonderful!! 🎉🎉🎉 I feel like I also learned something new with this question.

I have a hunch that you could probably also use the tag ID by doing:
```hbs
{{#checkIf this.id '==' 'TAGID'}}
```

Based on these articles:
- https://getpublii.com/dev/website-global-variable/#contentstructure
- https://getpublii.com/dev/tag-tags/

It's probably also possible to dynamically select which tag to use in the Post settings via the [Theme Settings API](https://getpublii.com/dev/theme-settings-api/)'s [Tags Dropdown selector](https://getpublii.com/dev/tags-dropdown/). I presume the [syntax](https://getpublii.com/dev/theme-settings-api/#howtoaccessoptionvaluesofthepostsettingsinthetheme) would be something like this (assuming the setting was named `tagDropdownSelection`):

```hbs
{{#checkIf this.id '==' '{{ @config.post.tagDropdownSelection }}'}}
```

**Reply by 3muchfun** [reply]
Thanks, @candidexmedia, using the tag name is fine for me, as as it easier to understand what the code does...
I tried to describe what I did, maybe you can have a look whether I missed something:
https://fluegge.net/ohnein/how-to-add-a-page-to-a-publii-website/

**Reply by candidexmedia** [reply]
Having a reference page is a great idea, and the info seems accurate.

If you want it to be discoverable by other people, I would just suggest making the title more descriptive, for example, "How to add a post as a page to a Publii website" could become "How to create a post template which fetches posts by tag" or "Adding a tag-specific post listing/directory inside a post" or something along those lines.

**Reply by 3muchfun** [reply]
done: https://fluegge.net/ohnein/how-to-create-a-post-template-which-fetches-posts-by-tag/

**Comment by joan-medina-roca** [comment]
I would create a post. Then create a post template and assign it to this post. Then make Publii display this post as the main page.

Now I would edit the template to show cards to the other posts.

I don't know if this is the solution you are looking for, but I think it's kind of close. You can see this implemented at my website's [lamardeswing.com](https://lamardeswing.com/) frontpage.

**Reply by 3muchfun** [reply]
Thank you, but that is not exactly, what I would like to achieve
